### PR TITLE
[TCSACR-261] Add a new guide for component manager

### PR DESCRIPTION
**Pull request by hjhun**
Signed-off-by: Hwankyu Jhun <h.jhun@samsung.com>

### Change Description ###
Add a new guide for component manager

### API Changes ###
 - ACR: TCSACR-261

### CODE REVIEW

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    ```
+
+2.  Operate on the context:
+    -   Get the component ID and application ID:
+
+        ```
+        string componentId = compoInfo.ComponentId;
+        string applicationId = compoInfo.ApplicationId;
+        ```
+    -   Check the type of the component:
```
Comment (surbhig12) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Please change to:

Verify the type of the component:

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        ```
+
+    -   Get the label and the icon path:
+
+        ```
+        string label = compoInfo.Label;
+        string iconPath = compoInfo.IconPath;
+        ```
+
+3.  Call the `GetInstalledComponentsAsync()` method of the [Tizen.Applications.ComponentBased.ComponentManager](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentManager.html) class and retrieve all components and print their information:
```
Comment (surbhig12) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Suggest rephrasing to:

3.  Call the `GetInstalledComponentsAsync()` method of the [Tizen.Applications.ComponentBased.ComponentManager](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentManager.html) class, and retrieve all components and print their information:

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        Log.Debug("Tag", "ComponentType: " + compoInfo.ComponentType.ToString());
+        Log.Debug("Tag", "isIconDisplayed: " + compoInfo.IsIconDisplayed.ToString());
+        Log.Debug("Tag", "isManagedByTaskManager: " + compoInfo.IsManagedByTaskManager.ToString());
+    }
+    ```
+
+
+## Related Information
+  * Dependencies
+    -   Tizen 5.5 and Higher
```
Comment (surbhig12) [inline docs/application/dotnet/guides/app-management/component-manager.md]: In #840, it is Tizen 5.5.
Please add the word "Higher" there or remove it from here, to maintain consistency.

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
+
+# Component Manager
+
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Please tell me what the toc file is.

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+The component manager provides information about installed and running components.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
+
+# Component Manager
+
+The component manager provides information about installed and running components. 
+
+The main features of the `Tizen.Applications.ComponentBased.ComponentManager` class include:
+
+-   Managing the running component context
+
+    For running components, you can retrieve the application running context and operate on it. You can [manage thec component running context](#manage_context) with the `Tizen.Applications.ComponentBased.ComponentManager` class.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+-   Getting the component information
+
+    For installed (but not necessarily running) componentss, you can retrieve information with the [Tizen.Applications.ComponentBased.ComponentInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentInfo.html) class.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+Iterator methods are used to travel through a list of components. The `GetRunningComponentsAsync()` method of the `Tizen.Applications.ComponentBased.ComponentManager` class is used for running components and the `GetInstalledComponentsAsync()` method is used for installed components.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+    ```
+    <privileges>
+       <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+       <privilege>http://tizen.org/privilege/packagemanager.info</privilege>
+    </privileges>
+    ```
+
+<a name="manage_context"></a>
+## Managing the Component Running Context
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+To get the component running context and its details, and to operate on the context:
+
+1.  Get the context of the currently-running component by creating an instance of the [Tizen.Applications.ComponentBased.ComponentRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentRunningContext.html) class, with the ID of the component from which the context is being obtained as a parameter.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+    When an component is not running, it is impossible to get its context.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+2.  Operate on the context:
+    -   Get the component ID, application ID, and instance ID from the context:
+
+        ```
+        string componentId = compoRunningContext.ComponentId;
+        string applicationId = compoRunningContext.ApplicationId;
+        string instanceId = compoRunningContext.InstanceId;
+        ```
+
+    -   Check the state of the component:
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+        ```
+
+    -   Resume the running application:
+
+        ```
+        compoRunningContext.Resume();
+        ```
+
+<a name="filter"></a>
+## Getting the Component Information
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+To get the installed information and its details, and to operate on the information:
+
+1.  Get the information of the currently-installed component by creating an instance of the [Tizen.Applications.ComponentBased.ComponentInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentInfo.html) class, with the ID of the component from which the information is being obtained as a parameter.
+
+    When an component is not installed, it is impossible to get its information.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: To get a component's information, the component must be installed.

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 1 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 2 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 3 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (surbhig12) [reply docs/application/dotnet/guides/app-management/component-manager.md]: The location of the toc_all.md file is:

docs/application/toc_all.md

[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 15]
```diff
@@ -0,0 +1,142 @@
+
+# Component Manager
+
+Component manager provides information about installed and running components.
+
+The main features of the `Tizen.Applications.ComponentBased.ComponentManager` class include:
+
+-   Managing running component context
+
+    For the running components, you can retrieve the component running context and operate on it. You can [manage thec component running context](#manage_context) with the `Tizen.Applications.ComponentBased.ComponentManager` class.
```
Comment (surbhig12) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Please correct the spelling mistake "thec":

You can [manage the component running context](#manage_context) with the ......

Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+    ```
+
+2.  Operate on the context:
+    -   Get the component ID and application ID:
+
+        ```
+        string componentId = compoInfo.ComponentId;
+        string applicationId = compoInfo.ApplicationId;
+        ```
+    -   Verify: the type of the component:
```
Comment (surbhig12) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Please remove the colon (:) after verify.

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+            /// The component is resumed.
+        else if (compoRunningContext.State == ComponentRunningContext.ComponentState.Paused)
+            /// The component is paused
+        else if (compoRunningContext.State == ComponentRunningContext.ComponentState.Destroyed)
+            /// The component is destroyed.
+        else
+            /// State is undefined
+        ```
+
+    -   Resume the running application:
```
Comment (surbhig12) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Is this step 3?

**Comment by hjhun** [comment]
If I should change any thing, please tell me.

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+2.  Operate on the context:
+    -   Get the component ID, application ID, and instance ID from the context:
+
+        ```
+        string componentId = compoRunningContext.ComponentId;
+        string applicationId = compoRunningContext.ApplicationId;
+        string instanceId = compoRunningContext.InstanceId;
+        ```
+
+    -   Verify the state of the component:
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/component-manager.md]: @surbhig12, would you please refer https://github.com/Samsung/tizen-docs/pull/840#discussion_r330375642?

To me it looks like the component states must be checked, not verified (as I saied, "verify" sound like "prove" to me).

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+
+    ```
+    <privileges>
+       <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+       <privilege>http://tizen.org/privilege/packagemanager.info</privilege>
+    </privileges>
+    ```
+
+<a name="manage_context"></a>
+## Managing Component Running Context
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Global comment.
It is minor issue. Isn't the "Managing Running Component Context" more accurate?

@hjhun, would you please check?

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+        ```
+        compoRunningContext.Resume();
+        ```
+
+<a name="filter"></a>
+## Getting Component Information
+
+To get the installed information and its details, and to operate on the information:
+
+1.  Get the information of the currently-installed component by creating an instance of the [Tizen.Applications.ComponentBased.ComponentInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentInfo.html) class, with the ID of the component from which the information is being obtained as a parameter.
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/component-manager.md]: with the ID of the component from which the information is being obtained as a parameter.

How about below?

with the ID of information obtained component as a parameter

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 21]
```diff
@@ -0,0 +1,142 @@
+
+# Component Manager
+
+Component manager provides information about installed and running components.
+
+The main features of the `Tizen.Applications.ComponentBased.ComponentManager` class include:
+
+-   Managing running component context
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Similar suggestion with  https://github.com/Samsung/tizen-docs/pull/840#discussion_r330386233.

@surbhig12 would you please check the comment and share your opinion with us?

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 22]
```diff
@@ -0,0 +1,142 @@
+
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 21 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+Iterator methods are used to travel through a list of components. The `GetRunningComponentsAsync()` method of the `Tizen.Applications.ComponentBased.ComponentManager` class is used for the running components and the `GetInstalledComponentsAsync()` method is used for the installed components.
+
+## Prerequisites
+
+To enable your application to use the component management functionality:
+
+1.  To use the methods and properties of the [Tizen.Applications.ComponentBased.ComponentManager](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ApplicationManager.html), [Tizen.Applications.ComponentBased.ComponentRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentRunningContext.html), and [Tizen.Applications.ComponentBased.ComponentInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentInfo.html) classes, include the [Tizen.Applications.ComponentBased](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.html) namespace in your application:
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 24]
```diff
@@ -0,0 +1,142 @@
...
+
+    ```
+    using Tizen.Applications.ComponentBased;
+    ```
+
+2.  To use classes and methods, the application has to request permission by adding the following privilege to the `tizen-manifest.xml` file:
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 19 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 25]
```diff
@@ -0,0 +1,142 @@
...
+       <privilege>http://tizen.org/privilege/packagemanager.info</privilege>
+    </privileges>
+    ```
+
+<a name="manage_context"></a>
+## Managing Component Running Context
+
+To get the component running context and its details, and to operate on the context:
+
+1.  Get the context of the currently running component by creating an instance of the [Tizen.Applications.ComponentBased.ComponentRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentRunningContext.html) class, with the ID of the component from which the context is being obtained as a parameter.
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 17 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: It's not.

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 20 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 16 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 26]
```diff
@@ -0,0 +1,142 @@
...
+        Log.Debug("Tag", "iconPath: " + compoInfo.IconPath);
+        Log.Debug("Tag", "ComponentType: " + compoInfo.ComponentType.ToString());
+        Log.Debug("Tag", "isIconDisplayed: " + compoInfo.IsIconDisplayed.ToString());
+        Log.Debug("Tag", "isManagedByTaskManager: " + compoInfo.IsManagedByTaskManager.ToString());
+    }
+    ```
+
+
+## Related Information
+  * Dependencies
```
[the remark this answers is not in the document]
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

**Comment by surbhig12** [comment]
@hjhun,

Request you to please rebase the PR to resolve the conflicts.

**Comment by hjhun** [comment]
> @hjhun,
> 
> Request you to please rebase the PR to resolve the conflicts.

Done

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 27]
```diff
@@ -0,0 +1,141 @@
...
+2.  To use the methods and properties of the [Tizen.Applications.ComponentBased.ComponentManager](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentManager.html), [Tizen.Applications.ComponentBased.ComponentRunningContext](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentRunningContext.html), and [Tizen.Applications.ComponentBased.ComponentInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentInfo.html) classes, include the [Tizen.Applications.ComponentBased](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.html) namespace in your application:
+
+    ```
+    using Tizen.Applications.ComponentBased;
+    ```
+
+<a name="manage_context"></a>
+## Managing Running Component Context
+
+To get the component running context and its details, and to operate on the context:
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/component-manager.md]: How about changing "component" and "running"?

To get the running component context and its details, and to operate on the context:

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 28]
```diff
@@ -0,0 +1,141 @@
...
+
+1.  Get the information of the currently-installed component by creating an instance of the [Tizen.Applications.ComponentBased.ComponentInfo](https://developer.tizen.org/dev-guide/csapi/api/Tizen.Applications.ComponentBased.ComponentInfo.html) class, with the ID of information obtained component as a parameter.
+
+    To get a component's information, the component must be installed.
+
+    ```
+    ComponentInfo compoInfo = new ComponentInfo(Your Component ID);
+    ```
+
+2.  Operate on the context:
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/app-management/component-manager.md]: Isn't "Operate on the information:" correct? Would you please check?

Comment (surbhig12) [reply docs/application/dotnet/guides/app-management/component-manager.md]: @mijin85cho, yes, it should be: Operate on the information

File: docs/application/dotnet/guides/app-management/component-manager.md [hunk 29]
```diff
@@ -0,0 +1,141 @@
+# Component Manager
+
+Component manager provides information about installed and running components.
+
+The main features of the `Tizen.Applications.ComponentBased.ComponentManager` class include:
+
+-   Managing running component context
+
+    For the running components, you can retrieve the component running context and operate on it. You can [manage the component running context](#manage_context) with the `Tizen.Applications.ComponentBased.ComponentManager` class.
```
[the remark this answers is not in the document]
Comment (surbhig12) [reply docs/application/dotnet/guides/app-management/component-manager.md]: @mijin85cho,
 In my opinion the current is fine. If we change, then I am afraid we won't b able to emphasis on "For the running components" and "For components that are installed but not necessarily running".
I suggest we keep it as it is as we are also maintaining parallelism within that section.

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 27 above)
Comment (surbhig12) [reply docs/application/dotnet/guides/app-management/component-manager.md]: @mijin85cho, thank you for the suggestion. I agree with it.

Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

File: docs/application/dotnet/guides/app-management/component-manager.md (hunk 28 above)
Comment (hjhun) [reply docs/application/dotnet/guides/app-management/component-manager.md]: Done

**Pull request merged**